ni_measurementlink_service/_internal/grpc_servicer.py

```diff
@@ -83,7 +83,28 @@ def time_remaining(self) -> float:
 
     def abort(self, code: grpc.StatusCode, details: str) -> None:
         """Aborts the RPC."""
-        self._grpc_context.abort(code, details)
+        try:
+            self._grpc_context.abort(code, details)
+        except Exception as e:
+            if isinstance(e, Exception):
+                raise CustomRpcError(code)
+            raise e
+
+
+class CustomRpcError(grpc.RpcError):
+    """A custom exception class for handling gRPC RPC errors.
+
+    gRPC's built-in RpcError is not directly configurable in Python, so this class
+    enables the creation of custom RPC errors with specific error codes.
+    """
+
+    def __init__(self, code: grpc.StatusCode) -> None:
+        """Initialize a CustomRpcError instance with a specific gRPC status code."""
+        self._code = code
+
+    def code(self) -> grpc.StatusCode:
+        """Get the gRPC status code associated with this custom RPC error."""
```

Comment:
More words are not always better.
```suggestion
        """Initialize a CustomRpcError instance."""
        self._code = code

    def code(self) -> grpc.StatusCode:
        """Get the gRPC status code."""
```